packages/gatsby-transformer-remark/src/extend-node-type.js

```diff
@@ -242,27 +251,53 @@ module.exports = (
       }
     }
 
-    async function getTableOfContents(markdownNode, pathToSlugField) {
+    async function getTableOfContents(markdownNode, gqlTocOptions) {
+      // fetch defaults
+      let appliedTocOptions = tocOptions
```

Comment:
```suggestion
      let appliedTocOptions = { ...tocOptions, ... gqlTocOptions }
```
and you won't need all those `if` clauses below